scanspec/regions.py

```diff
@@ -1,11 +1,13 @@
+from __future__ import annotations
+
 from dataclasses import dataclass, is_dataclass
-from typing import Iterator, List, Set
+from typing import Generic, Iterator, List, Set
 
 import numpy as np
 from apischema import schema
 from typing_extensions import Annotated as A
 
-from .core import AxesPoints, Serializable, if_instance_do
+from .core import AxesPoints, Axis, as_tagged_union, if_instance_do
```

Comment:
I did this because they are conceptually the same thing, the `Axis` for a `Region` is the same `Axis` for a `Spec` and `Frames`. This makes the docs read better as it all links to the same TypeVar. Would be worth having an in person discussion on this though...